File: HashMaps/205_isomorphic_strings/205_isomorphic_strings_soln.py
# 205. Isomorphic Strings

class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        maa = dict()
        mpp = dict()

        for i in range(len(s)):
            if (s[i] not in maa):
                maa[s[i]] = t[i]
            else:
                if maa[s[i]] != t[i]:
                    return False
            if t[i] not in mpp:
                mpp[t[i]] = s[i]
            else:
                if mpp[t[i]] != s[i]:
                    return False
        return True
        

File: HashMaps/205_isomorphic_strings/205_isomorphic_strings_soln.cpp
// 205. Isomorphic Strings

class Solution {
public:
    bool isIsomorphic(string s, string t) {
        //vector<int> map1(256,INT_MIN);
        int map1[256];
        //vector<int> map2(256,INT_MIN);
        int map2[256];
        fill(begin(map1), end(map1), INT_MIN);
        fill(begin(map2), end(map2), INT_MIN);
        int ti, si;
        for (int i=0; i<t.size(); i++) {
            
            ti = t[i];
            si = s[i];
            if (map1[ti] == INT_MIN) {
                map1[ti] = ti - si;
                cout << ti << "   " << si << '\n';
            }
            else {
                cout << ti << "   " << si << '\n';
                if (map1[ti] != (ti - si)) {
                    return false;
                }
            }
            if (map2[si] == INT_MIN) {
                map2[si] = ti - si;
                cout << ti << " ++  " << si << '\n';

            } else {
                cout << ti << "  -- " << si << '\n';
                if (map2[si] != ti - si) {
                    return false;
                }
            }
        }
        return true;
    }
};
